Sources/NetworkService/NetworkError/NetworkError.swift

```diff
@@ -0,0 +1,32 @@
+import Foundation
+
+/// Contains network errors that may occur, while the application is running.
+public enum NetworkError: Error, CaseIterable {
```

Comment:
As I said earlier if you use the `CaseIterable` only in the test target, then implement it within the test target.

Reply:
Done.